src/maestral/main.py

```diff
@@ -865,6 +882,99 @@ def list_revisions(self, dbx_path: str, limit: int = 10) -> List[StoneType]:
 
+    def get_file_diff(
+        self, dbx_path: str, old_rev: str, new_rev: Optional[str] = None
+    ) -> List[str]:
+        """
+        Download the two revisions if necessary and create a
+        diff with the python difflib library. If new_rev is None,
+        it will compare the old revision to the current file.
+
+        :param old_rev: Hash of old revision.
+        :param new_rev: Hash of new revision.
+        :returns: Diff as a list of lines.
+        :raises UnsupportedFileTypeForDiff: if file type is not supported.
+        :raises UnsupportedFileTypeForDiff: if file content could not be decoded.
+        :raises MaestralApiError: if file could not be read for any other reason.
+        """
+
+        def pretty_date(s: datetime) -> str:
+            """
+            Prettify the 'client_modified' metadata.
+            """
+
+            return s.astimezone().strftime("%d %b %Y at %H:%M")
+
+        def download_rev(f: IO[bytes], rev: str) -> Tuple[str, str]:
+            """
+            Download a rev to 'f' and return the location and the date.
+            """
+
+            location = f.name
+            date = pretty_date(
+                self.client.download(dbx_path, location, rev=rev).client_modified
+            )
+            return (location, date)
+
+        all_revs = self.client.list_revisions(dbx_path).entries
+        full_path = self.sync.to_local_path(dbx_path)
+
+        # Check if a diff is possible
+        # If mime is None, procede because most files without
+        # an extension are just text files
+        mime, _ = mimetypes.guess_type(full_path)
+        if mime is not None and not mime.startswith("text/"):
+            raise UnsupportedFileTypeForDiff(
+                f"Bad file type: '{mime}'", "Only files of type 'text/*' are supported."
+            )
+
+        # if new_rev is None, make it to the newest one
+        if new_rev is None:
+            new_rev = all_revs[0].rev
+        new_location = full_path
+        new_date = pretty_date(all_revs[0].client_modified)
+
+        # Check if the revision is the newest
+        # and see if it is avaible locally; download it if not
+        if new_rev is not all_revs[0].rev:
+            tmp_n = tempfile.NamedTemporaryFile()
+            new_location, new_date = download_rev(tmp_n, new_rev)
+
+        tmp_o = tempfile.NamedTemporaryFile()
+        old_location, old_date = download_rev(tmp_o, old_rev)
+
+        try:
+            with convert_api_errors(dbx_path=dbx_path, local_path=new_location):
+                try:
+                    with open(new_location) as f:
+                        new_content = f.readlines()
+                # If the file was not found, retry download once
+                # Possible if the file is only stored in the cloud
+                except FileNotFoundError:
+                    tmp_n = tempfile.NamedTemporaryFile()
+                    new_location, new_date = self.client.download(tmp_n, new_rev)
```

Comment:
Type error: you need to pass the file name `tmp_n.name` to the download function. Also, surely, you meant to use `download_rev`?

Reply:
Sorry, I was a little tired yesterday :(

Reply:
No worries, this happens to all of us!